File: src/0896.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        return isDec(nums) || isAsc(nums);
    }

    bool isDec(vector<int> n) {
        for(int i = 0; i < n.size() - 1; ++i) {
            if(n[i] < n[i + 1]) {
                return false;
            }
        }
        return true;
    }
    
    bool isAsc(vector<int> n) {
        for(int i = 0; i < n.size() - 1; ++i) {
            if(n[i] > n[i + 1]) {
                return false;
            }
        }
        return true;
    }
};

File: src/0896.py
class Solution:
    def isMonotonic(self, nums: list[int]) -> bool:
        def isDec(n):
            for i in range(len(n) - 1):
                if n[i] < n[i + 1]:
                    return False
            return True
        
        def isAsc(n):
            for i in range(len(n) - 1):
                if n[i] > n[i + 1]:
                    return False
            return True

        return isAsc(nums) or isDec(nums)
